clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -347,9 +432,18 @@ public int sessionId() {
                 partitionResponses.add(partitionData.partitionResponse);
                 topicResponseList.add(new FetchResponseData.FetchableTopicResponse()
                         .setTopic(entry.getKey().topic())
+                        .setTopicId(topicIds.getOrDefault(entry.getKey().topic(), Uuid.ZERO_UUID))
                         .setPartitionResponses(partitionResponses));
             }
         });
+        // ID errors will be empty unless topic IDs are supported and there were topic ID errors
```

Comment:
It will be good to see if can separate out new and old forms of FetchRequest/Response. It is not a big deal since it is just wrapping the protocol layer.